From a57d295c52cb04659081f050a119fe2f78368fb3 Mon Sep 17 00:00:00 2001
From: Albert I <kras@raphielgang.org>
Date: Fri, 19 Feb 2021 18:46:51 +0800
Subject: [PATCH 1/2] Revert "frameworks/base: Support for third party NFC
 features and extensions"

This reverts commit fe9be1bdf5d1188a6f7bb4a434c7fd56754f85c3.

Breaks build with AOSP NFC stack.

Signed-off-by: Albert I <kras@raphielgang.org>
Change-Id: I4b94b1b1ee1813d4ee4f508b4cd3a745e29ed52a
---
 core/java/android/nfc/INfcAdapter.aidl        |  6 -----
 .../android/nfc/cardemulation/AidGroup.java   | 11 ++++-----
 .../nfc/cardemulation/ApduServiceInfo.java    | 23 ++++++++-----------
 core/java/android/nfc/tech/MifareClassic.java |  6 -----
 core/java/android/nfc/tech/NfcA.java          | 13 ++---------
 5 files changed, 16 insertions(+), 43 deletions(-)

diff --git a/core/java/android/nfc/INfcAdapter.aidl b/core/java/android/nfc/INfcAdapter.aidl
index c758031aacf..0b2cfdd9ece 100644
--- a/core/java/android/nfc/INfcAdapter.aidl
+++ b/core/java/android/nfc/INfcAdapter.aidl
@@ -1,7 +1,4 @@
 /*
- * Copyright (c) 2018, The Linux Foundation. All rights reserved.
- * Not a Contribution.
- *
  * Copyright (C) 2010 The Android Open Source Project
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
@@ -34,7 +31,6 @@ import android.nfc.INfcUnlockHandler;
 import android.nfc.ITagRemovedCallback;
 import android.nfc.INfcDta;
 import android.os.Bundle;
-import android.os.IBinder;
 
 /**
  * @hide
@@ -46,8 +42,6 @@ interface INfcAdapter
     INfcFCardEmulation getNfcFCardEmulationInterface();
     INfcAdapterExtras getNfcAdapterExtrasInterface(in String pkg);
     INfcDta getNfcDtaInterface(in String pkg);
-    IBinder getNfcAdapterVendorInterface(in String vendor);
-
     int getState();
     boolean disable(boolean saveState);
     boolean enable();
diff --git a/core/java/android/nfc/cardemulation/AidGroup.java b/core/java/android/nfc/cardemulation/AidGroup.java
index d4c966be04b..2436e57b74b 100644
--- a/core/java/android/nfc/cardemulation/AidGroup.java
+++ b/core/java/android/nfc/cardemulation/AidGroup.java
@@ -1,7 +1,4 @@
 /*
- * Copyright (c) 2018, The Linux Foundation. All rights reserved.
- * Not a Contribution.
- *
  * Copyright (C) 2015 The Android Open Source Project
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
@@ -42,7 +39,7 @@ import java.util.List;
  *
  * @hide
  */
-public class AidGroup implements Parcelable {
+public final class AidGroup implements Parcelable {
     /**
      * The maximum number of AIDs that can be present in any one group.
      */
@@ -51,11 +48,11 @@ public class AidGroup implements Parcelable {
     static final String TAG = "AidGroup";
 
     @UnsupportedAppUsage
-    protected List<String> aids;
+    final List<String> aids;
     @UnsupportedAppUsage
-    protected String category;
+    final String category;
     @UnsupportedAppUsage
-    protected String description;
+    final String description;
 
     /**
      * Creates a new AidGroup object.
diff --git a/core/java/android/nfc/cardemulation/ApduServiceInfo.java b/core/java/android/nfc/cardemulation/ApduServiceInfo.java
index 1cdf14f8cc8..d7c2e0522b0 100644
--- a/core/java/android/nfc/cardemulation/ApduServiceInfo.java
+++ b/core/java/android/nfc/cardemulation/ApduServiceInfo.java
@@ -1,7 +1,4 @@
 /*
- * Copyright (c) 2018, The Linux Foundation. All rights reserved.
- * Not a Contribution.
- *
  * Copyright (C) 2013 The Android Open Source Project
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
@@ -51,24 +48,24 @@ import java.util.Map;
 /**
  * @hide
  */
-public class ApduServiceInfo implements Parcelable {
+public final class ApduServiceInfo implements Parcelable {
     static final String TAG = "ApduServiceInfo";
 
     /**
      * The service that implements this
      */
     @UnsupportedAppUsage
-    protected ResolveInfo mService;
+    final ResolveInfo mService;
 
     /**
      * Description of the service
      */
-    protected String mDescription;
+    final String mDescription;
 
     /**
      * Whether this service represents AIDs running on the host CPU
      */
-    protected boolean mOnHost;
+    final boolean mOnHost;
 
     /**
      * Offhost reader name.
@@ -86,33 +83,33 @@ public class ApduServiceInfo implements Parcelable {
      * Mapping from category to static AID group
      */
     @UnsupportedAppUsage
-    protected HashMap<String, AidGroup> mStaticAidGroups;
+    final HashMap<String, AidGroup> mStaticAidGroups;
 
     /**
      * Mapping from category to dynamic AID group
      */
     @UnsupportedAppUsage
-    protected HashMap<String, AidGroup> mDynamicAidGroups;
+    final HashMap<String, AidGroup> mDynamicAidGroups;
 
     /**
      * Whether this service should only be started when the device is unlocked.
      */
-    protected boolean mRequiresDeviceUnlock;
+    final boolean mRequiresDeviceUnlock;
 
     /**
      * The id of the service banner specified in XML.
      */
-    protected int mBannerResourceId;
+    final int mBannerResourceId;
 
     /**
      * The uid of the package the service belongs to
      */
-    protected int mUid;
+    final int mUid;
 
     /**
      * Settings Activity for this service
      */
-    protected String mSettingsActivityName;
+    final String mSettingsActivityName;
 
     /**
      * @hide
diff --git a/core/java/android/nfc/tech/MifareClassic.java b/core/java/android/nfc/tech/MifareClassic.java
index 9cae043c4bd..080e058737b 100644
--- a/core/java/android/nfc/tech/MifareClassic.java
+++ b/core/java/android/nfc/tech/MifareClassic.java
@@ -1,6 +1,4 @@
 /*
- * Copyright (C) 2018 NXP Semiconductors
- * The original Work has been changed by NXP Semiconductors.
  * Copyright (C) 2010 The Android Open Source Project
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
@@ -175,10 +173,6 @@ public final class MifareClassic extends BasicTagTechnology {
             mType = TYPE_CLASSIC;
             mSize = SIZE_4K;
             break;
-        case 0x19:
-            mType = TYPE_CLASSIC;
-            mSize = SIZE_2K;
-            break;
         case 0x28:
             mType = TYPE_CLASSIC;
             mSize = SIZE_1K;
diff --git a/core/java/android/nfc/tech/NfcA.java b/core/java/android/nfc/tech/NfcA.java
index 819e9e31b6e..88730f9af3d 100644
--- a/core/java/android/nfc/tech/NfcA.java
+++ b/core/java/android/nfc/tech/NfcA.java
@@ -1,6 +1,4 @@
 /*
- * Copyright (C) 2018 NXP Semiconductors
- * The original Work has been changed by NXP Semiconductors.
  * Copyright (C) 2010 The Android Open Source Project
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
@@ -68,15 +66,8 @@ public final class NfcA extends BasicTagTechnology {
     /** @hide */
     public NfcA(Tag tag) throws RemoteException {
         super(tag, TagTechnology.NFC_A);
-        Bundle extras;
-        mSak = 0;
-        if(tag.hasTech(TagTechnology.MIFARE_CLASSIC))
-        {
-            extras = tag.getTechExtras(TagTechnology.MIFARE_CLASSIC);
-            mSak = extras.getShort(EXTRA_SAK);
-        }
-        extras = tag.getTechExtras(TagTechnology.NFC_A);
-        mSak |= extras.getShort(EXTRA_SAK);
+        Bundle extras = tag.getTechExtras(TagTechnology.NFC_A);
+        mSak = extras.getShort(EXTRA_SAK);
         mAtqa = extras.getByteArray(EXTRA_ATQA);
     }
 
-- 
2.30.1


From 68dc3c0e2875b249627190922e4224e4e8cb9e95 Mon Sep 17 00:00:00 2001
From: Albert I <kras@raphielgang.org>
Date: Thu, 11 Feb 2021 11:49:50 +0800
Subject: [PATCH 2/2] zygote: Don't make seccomp depend on SELinux enforcing

This change was originally submitted to AOSP as c/1571580, but the change has
since been removed from Gerrit. This is necessary to avoid privilege escalation
that makes use of privileges that won't be dropped if SELinux is set.

RIP authorship, but I don't care for now.

Signed-off-by: Albert I <kras@raphielgang.org>
Change-Id: Iad3a929e9ad9daf82d9d817740b53d1757b53cbe
---
 core/jni/com_android_internal_os_Zygote.cpp | 16 ----------------
 1 file changed, 16 deletions(-)

diff --git a/core/jni/com_android_internal_os_Zygote.cpp b/core/jni/com_android_internal_os_Zygote.cpp
index 9eede83e21e..be00f6383e5 100644
--- a/core/jni/com_android_internal_os_Zygote.cpp
+++ b/core/jni/com_android_internal_os_Zygote.cpp
@@ -127,8 +127,6 @@ static jclass gZygoteClass;
 static jmethodID gCallPostForkSystemServerHooks;
 static jmethodID gCallPostForkChildHooks;
 
-static bool gIsSecurityEnforced = true;
-
 /**
  * True if the app process is running in its mount namespace.
  */
@@ -632,11 +630,6 @@ static void PreApplicationInit() {
 }
 
 static void SetUpSeccompFilter(uid_t uid, bool is_child_zygote) {
-  if (!gIsSecurityEnforced) {
-    ALOGI("seccomp disabled by setenforce 0");
-    return;
-  }
-
   // Apply system or app filter based on uid.
   if (uid >= AID_APP_START) {
     if (is_child_zygote) {
@@ -2198,11 +2191,6 @@ static void com_android_internal_os_Zygote_nativeAllowFileAcrossFork(
 
 static void com_android_internal_os_Zygote_nativeInstallSeccompUidGidFilter(
         JNIEnv* env, jclass, jint uidGidMin, jint uidGidMax) {
-  if (!gIsSecurityEnforced) {
-    ALOGI("seccomp disabled by setenforce 0");
-    return;
-  }
-
   bool installed = install_setuidgid_seccomp_filter(uidGidMin, uidGidMax);
   if (!installed) {
       RuntimeAbort(env, __LINE__, "Could not install setuid/setgid seccomp filter.");
@@ -2280,10 +2268,6 @@ static void com_android_internal_os_Zygote_nativeInitNativeState(JNIEnv* env, jc
    * Security Initialization
    */
 
-  // security_getenforce is not allowed on app process. Initialize and cache
-  // the value before zygote forks.
-  gIsSecurityEnforced = security_getenforce();
-
   selinux_android_seapp_context_init();
 
   /*
-- 
2.30.1

